Crash report 3.6: ArrayIndexOutOfBoundsException in StatsVisitorsAndViewsFragment

```
java.lang.RuntimeException: Unable to resume activity {org.wordpress.android/org.wordpress.android.ui.stats.StatsActivity}: java.lang.ArrayIndexOutOfBoundsException: length=7; index=9
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3076)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3105)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
       at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4054)
       at android.app.ActivityThread.access$1000(ActivityThread.java:175)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1314)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:146)
       at android.app.ActivityThread.main(ActivityThread.java:5602)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1283)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1099)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by: java.lang.ArrayIndexOutOfBoundsException: length=7; index=9
       at org.wordpress.android.ui.stats.StatsVisitorsAndViewsFragment.updateUIBelowTheGraph(StatsVisitorsAndViewsFragment.java:298)
       at org.wordpress.android.ui.stats.StatsVisitorsAndViewsFragment.updateUI(StatsVisitorsAndViewsFragment.java:281)
       at org.wordpress.android.ui.stats.StatsVisitorsAndViewsFragment.onResume(StatsVisitorsAndViewsFragment.java:190)
       at android.app.Fragment.performResume(Fragment.java:1743)
       at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:924)
       at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1062)
       at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1044)
       at android.app.FragmentManagerImpl.dispatchResume(FragmentManager.java:1869)
       at android.app.Activity.performResume(Activity.java:5540)
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3066)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3105)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
       at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4054)
       at android.app.ActivityThread.access$1000(ActivityThread.java:175)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1314)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:146)
       at android.app.ActivityThread.main(ActivityThread.java:5602)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1283)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1099)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

--- a/WordPress/src/main/java/org/wordpress/android/ui/stats/StatsUIHelper.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/stats/StatsUIHelper.java
@@ -39,10 +39,9 @@ public class StatsUIHelper {
         return (point.y < point.x);
     }
 
-    // Load more bars for 720DP tablets and 600DP tablets in landscape
-    public static boolean shouldLoadMoreBars(Activity act) {
-        return (StatsUtils.getSmallestWidthDP() >= TABLET_720DP
-                || (StatsUtils.getSmallestWidthDP() == TABLET_600DP && isInLandscape(act)));
+    // Load more bars for 720DP tablets
+    public static boolean shouldLoadMoreBars() {
+        return (StatsUtils.getSmallestWidthDP() >= TABLET_720DP);
     }
 
     public static void reloadLinearLayout(Context ctx, ListAdapter adapter, LinearLayout linearLayout, int maxNumberOfItemsToshow) {

--- a/WordPress/src/main/java/org/wordpress/android/ui/stats/StatsVisitorsAndViewsFragment.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/stats/StatsVisitorsAndViewsFragment.java
@@ -275,10 +275,23 @@ public class StatsVisitorsAndViewsFragment extends StatsAbstractFragment
         mGraphView.setHorizontalLabels(horLabels);
         mGraphView.setGestureListener(this);
 
-        int barSelectedOnGraph = mSelectedBarGraphBarIndex != -1 ? mSelectedBarGraphBarIndex : dataToShowOnGraph.length - 1;
-        mGraphView.highlightBar(barSelectedOnGraph);
+        int barSelectedOnGraph;
+        if (mSelectedBarGraphBarIndex == -1) {
+            // No previous bar was highlighted, highlight the most recent one
+            barSelectedOnGraph = dataToShowOnGraph.length - 1;
+        } else if (mSelectedBarGraphBarIndex < dataToShowOnGraph.length) {
+            barSelectedOnGraph = mSelectedBarGraphBarIndex;
+        } else {
+            // A previous bar was highlighted, but it's out of the screen now. Device Rotated.
+            // This cannot happen now, since we've fixed number of bars on a device. # of bars doesn't change with device rotation.
+            barSelectedOnGraph = dataToShowOnGraph.length - 1;
+            mSelectedBarGraphBarIndex = barSelectedOnGraph;
+            // TODO: make sure to handle this case in the modules below, otherwise the graph is updated but not other fragments
+            // that are still pointing to the old selected date.
+        }
 
         updateUIBelowTheGraph(barSelectedOnGraph);
+        mGraphView.highlightBar(barSelectedOnGraph);
     }
 
     //update the area right below the graph
@@ -294,6 +307,12 @@ public class StatsVisitorsAndViewsFragment extends StatsAbstractFragment
         }
 
         final VisitModel[] dataToShowOnGraph = getDataToShowOnGraph((VisitsModel)mVisitsData);
+
+        // This check should never be true, since we put a check on the index in the calling function updateUI()
+        if (dataToShowOnGraph.length <= itemPosition) {
+            // Make sure we're not highlighting
+            itemPosition = dataToShowOnGraph.length -1;
+        }
 
         String date =  mStatsDate[itemPosition];
         if (date == null) {
@@ -427,7 +446,7 @@ public class StatsVisitorsAndViewsFragment extends StatsAbstractFragment
             return 0;
         }
 
-        if(StatsUIHelper.shouldLoadMoreBars(getActivity())) {
+        if(StatsUIHelper.shouldLoadMoreBars()) {
             return 10;
         } else {
             return 7;

